package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs

```diff
@@ -603,55 +676,120 @@ public class GeometrySubsets {
 #endif
     }
 
-    static void LoadPrimvars(Scene scene,
-                             Mesh unityMesh,
-                             string usdMeshPath,
-                             List<string> primvars,
-                             int[] faceVertexCounts,
-                             int[] faceVertexIndices) {
-      if (primvars == null || primvars.Count == 0) { return; }
+    private static Vector3[] SanitizePointsForUnity(Vector3[] vertices, int[] indices)
+    {
+      int indicesCount = indices.Length;
+      Vector3[] sanitizedPoints = new Vector3[indicesCount];
+
+      for (int i = 0; i < indicesCount; i++)
+      {
+        int vertexIndex = indices[i];
+        sanitizedPoints[i] = vertices[vertexIndex];
+      }
+
+      return sanitizedPoints;
+    }
+
+    private static bool IsThereFaceVaryingAttributes(MeshSample usdMesh, SceneImportOptions options)
+    {
+      // If an attribute is face varying, its number of element will be usdMesh.indices.Length.
+      int pointCount = usdMesh.points.Length;
+      int faceCount = usdMesh.faceVertexCounts.Length;
+
+      // Potential face varying attributes are: normals, tangents, color and uv (st, uv, uv2. uv3. uv4).
```

Comment:
Any primvar requested by UsdPreviewSurface also needs to be included....

Reply:
The `|| options.ShouldBindMaterials` is here for that... It is a dirty shortcut... but I'm not sure how to check all the primvar requested by UsdPreviewSurface?